vayesta/core/qemb/fragment.py

```diff
@@ -1013,3 +1013,24 @@ def plot3d(self, filename, gridsize=(100, 100, 100), **kwargs):
         cube = CubeFile(self.mol, filename=filename, nx=nx, ny=ny, nz=nz, **kwargs)
         cube.add_orbital(self.c_frag)
         cube.write()
+
+    def check_solver(self, solver):
+        is_uhf = np.ndim(self.base.mo_coeff[1]) == 2
+        if self.opts.screening:
+            is_eb = "crpa_full" in self.opts.screening
+        else:
+            is_eb = False
+        check_solver_config(is_uhf, is_eb, solver, self.log)
+
+    def get_solver(self, solver=None):
+        # This detects based on fragment what kind of Hamiltonian is appropriate (restricted and/or EB).
+        cl_ham = ClusterHamiltonian(self, self.mf, self.log, screening=self.opts.screening)
```

Comment:
It might be useful to keep the Hamiltonian outside the cluster solver scope, i.e. to generate it first?
This would help in cases where you want to keep the cluster Hamiltonian stored after the solver completed

Reply:
This could be nice, though if we cache the ERIs within it obviously need to ensure it's deleted to avoid using too much memory, as usual. This would then allow the energy calculation to access whatever eris are needed directly, potentially making use of cderis if available.